### ISSUE TITLE
Prevent SolrIndexJob from being manually run in the CMS

### ISSUE THREAD

**Issue by phptek** [Firesphere/silverstripe-solr-search#174]
**Describe the bug**

Creating and attempting to run the `SolrIndexJob` in the CMS' "Jobs" section will result in errors becuase PCNTL is required by the `SolrIndexTask` which is never going to be available in any non-CLI, PHP SAPI.

**To Reproduce**

* Login to the CMS
* Select SolrIndexJob from the "Job ot create" dropdown in the "Jobs" admin
* Attempt to run it
* Observe error in browser's network tab

**Expected behavior**

SolrIndexJob should not appear with a "Run" icon in the Jobs GridField. It is only invoked by `ProcessJobQueTask` via the CLI

**Comment by phptek** [comment]
Related note, the existing `SolrIndexJob` does not specify a job-type, and as such will be run as frequently as the standard `ProcessJobQueueTask` which on our system is set as `* * * * *`. For my current project we're indexing around 1500 pages and ~10Gb of uploaded PDFs, and the `SolrIndexTask` takes arond ~20m to run.

Therefore the `SolrIndexJob` should specify an appropriate `getJobType` method ala:

```
    /**
     * By default jobs should just go into the default processing queue
     *
     * @return string
     */
    public function getJobType()
    {
        return QueuedJob::LARGE;
    }
```

**Comment by Firesphere** [comment]
It should already have the flag to only run in CLI where PCNTL is available? :confused:

**Comment by Firesphere** [comment]
Bah, humbug, other way around, only run PCNTL in CLI, not when not in CLI mode

**Comment by Firesphere** [comment]
On another note...

The index job should run a very select set of groups, not all groups at once, so the job should not timeout from CLI. If this is happening, please make another bug report for that?

**Comment by phptek** [comment]
> It should already have the flag to only run in CLI where PCNTL is available? confused

I am merely reporting what I have observed; which is to say: I can opt to run the `SolrIndexJob` from the Queued Jobs GridField - whch I believe users shouldn't be able to do. Becuase the job fires off `/dev/tasks/SolrIndexTask` while still using the FPM SAPI not the CLI, then you get the error:

![Screenshot](https://user-images.githubusercontent.com/478440/71204208-72bc3600-2304-11ea-8eb3-420251e88d3c.png)

From: https://www.php.net/manual/en/intro.pcntl.php

> "Process Control should not be enabled within a web server environment and unexpected results may happen if any Process Control functions are used within a web server environment."

**Comment by Firesphere** [comment]
Hmmmm... I'm really sure I fixed this up before already. You're on beta 3? It may be that I fixed that issue _after_ that release...

**Comment by phptek** [comment]
dev-master as of this morning. I'll double check again

**Comment by Firesphere** [comment]
Thanks for the reporting by the way. Helps fixing up stuff a lot :)

**Comment by phptek** [comment]
```
vagrant@box: (develop-ss4) $> composer info firesphere/solr-search  | grep -A4 versions
versions : * dev-master, 1.0.x-dev
type     : silverstripe-vendormodule
license  : BSD 3-Clause "New" or "Revised" License (BSD-3-Clause) (OSI approved) https://spdx.org/licenses/BSD-3-Clause.html#licenseText
source   : [git] https://github.com/Firesphere/silverstripe-solr-search.git cf30204a5cd7033c14c9aac33bb4dc686664afc5
dist     : [zip] https://api.github.com/repos/Firesphere/silverstripe-solr-search/zipball/cf30204a5cd7033c14c9aac33bb4dc686664afc5 cf30204a5cd7033c14c9aac33bb4dc686664afc5
```

**Comment by Firesphere** [comment]
Still having the same problem?

**Comment by phptek** [comment]
> Still having the same problem?

Yup :-\

**Comment by Firesphere** [comment]
Bugger.... So, somehow `pcntl` is detected as true, even when not using the CLI...

That's annoying :/

**Comment by Firesphere** [comment]
Problem found... My change to fix the issue was not pushed to git. Solved in #182
